File: src/modules/health/usecases/health_check_usecase.py
from typing import Annotated
from fastapi import Depends
from src.modules.health.presenter.dtos import HealthCheckResponseDto


class HealthCheckUsecase():
    async def execute(self) -> HealthCheckResponseDto:
        return HealthCheckResponseDto(status="healthy")


HealthCheckUsecaseDep = Annotated[HealthCheckUsecase, Depends(HealthCheckUsecase)]

File: src/modules/health/usecases/health-check.usecase.ts
import { Injectable } from '@nestjs/common';
import { HealthCheckResponseDto } from '../presenter/dtos/health-check.dto';

@Injectable()
export class HealthCheckUsecase {
  execute(): Promise<HealthCheckResponseDto> {
    return Promise.resolve({ status: 'healthy' });
  }
}
